source/features/select-notifications.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+import {
+	CheckCircleIcon,
+	CheckIcon,
+	DotFillIcon,
+	DotIcon,
+	GitMergeIcon,
+	GitPullRequestDraftIcon,
+	GitPullRequestIcon,
+	IssueOpenedIcon,
+	XCircleIcon
+} from '@primer/octicons-react';
+
+import features from '.';
+
+const filters = {
+	'Pull requests': ':is(.octicon-git-pull-request, .octicon-git-pull-request-closed, .octicon-git-pull-request-draft, .octicon-git-merge)',
+	Issues: ':is(.octicon-issue-opened, .octicon-issue-closed)',
+	Open: ':is(.octicon-issue-opened, .octicon-git-pull-request)',
+	Closed: ':is(.octicon-issue-closed, .octicon-git-pull-request-closed)',
+	Draft: '.octicon-git-pull-request-draft',
+	Merged: '.octicon-git-merge',
+	Read: '.notification-read',
+	Unread: '.notification-unread'
+};
+
+type Filter = keyof typeof filters;
+type Category = 'Type' | 'Status' | 'Read';
+
+function resetFilters({target}: Event): void {
+	select('form#rgh-select-notifications-form')!.reset();
+	for (const label of select.all('label', target as Element)) {
+		label.setAttribute('aria-checked', 'false');
+	}
+}
+
+function handleSelection({target}: Event): void {
+	const selectAllCheckbox = select('input[type="checkbox"].js-notifications-mark-all-prompt')!;
+	// Reset the "Select all" checkbox
+	if (selectAllCheckbox.checked) {
+		selectAllCheckbox.click();
+	}
+
+	if (select.exists(':checked', target as Element)) {
+		const formData = new FormData(select('form#rgh-select-notifications-form'));
+		const types = formData.getAll('Type').map(type => filters[type as Filter]);
+		const statuses = formData.getAll('Status').map(status => filters[status as Filter]);
+		const readStatus = formData.getAll('Read').map(read => filters[read as Filter]);
+
+		for (const notification of select.all('.notifications-list-item')) {
+			if (
+				(types.length > 0 && !select.exists(types, notification)) ||
```

Comment:
Let's dry it up a bit

```js
function getSelector(formData, type): string {
	return formData.getAll(type).map(type => filters[type as Filter]).join(',')
}
```

```suggestion
		const readStatus = getSelector(formData, 'Read');

		for (const notification of select.all('.notifications-list-item')) {
			if (
				(types && !select.exists(types, notification)) ||
```